site/docs/install-compile-source.md

```diff
@@ -50,19 +50,22 @@ Ensure you have installed:
 *   **JDK 8.** You must install version 8 of the JDK. Versions other than 8 are
     *not* supported.
 
-*   **Python**. Versions 2 and 3 are supported. You *must* install the
+*   **Python**. Versions 2 and 3 are supported. You need the
     Windows-native version (downloadable from [https://www.python.org](https://www.python.org)).
     Versions installed via pacman in MSYS2 will not work.
 
-### 2.  Download and unpack Bazel's distribution archive
+### 2.  Download and unpack Bazel's source files (distribution archive)
 
-Download `bazel-<version>-dist.zip` from the [release page](https://github.com/bazelbuild/bazel/releases).
+Download `bazel-<version>-dist.zip` from [GitHub](https://github.com/bazelbuild/bazel/releases),
+e.g. `bazel-0.15.2-dist.zip`.
 
 **Note:** There is a **single, architecture-independent** distribution archive. There are no architecture-specific or OS-specific distribution archives.
 
-We recommend to also verify the signature made by our [release key](https://bazel.build/bazel-release.pub.gpg) 48457EE0.
+**Note:** You have to use the distribution archive to build Bazel from source.
+You cannot use a source tree cloned from GitHub.
```

Comment:
Hmm, how come a distribution archive needs to be used?

Is there a difference between the source code in the archive and the source code in the source tree that warrants needing to build from the archive? If so, what is this difference, and how can one get the set of files in the archive from just the source tree?

Reply:
Answered the first question in the doc.

As for the second: I feel reluctant to document this because I see no immediate need for it from the user's perspective. FYI, you can build it with `bazel build //:bazel-distfile`.

Reply:
Thanks @laszlocsomor!

I suppose my fear with not documenting this is that it would make things harder for people who wish to bootstrap Bazel from the Bazel Git repo rather than an archive. But then again, the archive contains all the needed source code to bootstrap Bazel, so my fear is probably unfounded. :)

Reply:
Right, once you can bootstrap from a release's distribution archive, you can build from HEAD using that Bazel.